pkg/cloudprovider/aws/instance.go

```diff
@@ -242,20 +247,24 @@ func (p *InstanceProvider) getInstances(ctx context.Context, ids []*string) ([]*
 	return instances, err
 }
 
-func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) (*v1.Node, error) {
+func (p *InstanceProvider) instanceToNode(ctx context.Context, constraints *v1alpha1.Constraints, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) (*v1.Node, error) {
 	for _, instanceType := range instanceTypes {
 		if instanceType.Name() == aws.StringValue(instance.InstanceType) {
 			nodeName := strings.ToLower(aws.StringValue(instance.PrivateDnsName))
 			if injection.GetOptions(ctx).GetAWSNodeNameConvention() == options.ResourceName {
 				nodeName = aws.StringValue(instance.InstanceId)
 			}
+
+			capacityType := getCapacityType(instance)
+			allocationStrategy, _ := p.getAllocationStrategy(constraints, capacityType)
 			return &v1.Node{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: nodeName,
 					Labels: map[string]string{
-						v1.LabelTopologyZone:       aws.StringValue(instance.Placement.AvailabilityZone),
-						v1.LabelInstanceTypeStable: aws.StringValue(instance.InstanceType),
-						v1alpha5.LabelCapacityType: getCapacityType(instance),
+						v1.LabelTopologyZone:             aws.StringValue(instance.Placement.AvailabilityZone),
+						v1.LabelInstanceTypeStable:       aws.StringValue(instance.InstanceType),
+						v1alpha5.LabelCapacityType:       capacityType,
+						v1alpha5.LabelAllocationStrategy: allocationStrategy,
```

Comment:
I don't think this is quite the right way to implement this as the allocation strategy is not a property of the node, it's more of a configuration option for the provisioner. I also can't imagine wanting a pod to only be on a "lowest-price" node, for example.

Reply:
I agree that the allocation strategy is not a property of the node so it seems odd to use it as a node label. However allocation strategy has an effect on the lifecycle of the node, so it seems to me possible there are some cases in which the users want their workload only in 'lowest-price` node or `capacity-optimized` node.